### ISSUE TITLE
remove "," as last item of initialization list

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/spartan-refactoring#74]
as in int[] a = new int[] {2,3,}; check also int [] a = new int[] {,}

**Comment by yossigil** [comment]
This should be easy. Please try to finish it early. Can be done in half a day.

**Comment by yossigil** [comment]
Good job!

**Comment by NivShalmon** [comment]
@yossigil We tried to run tests for the wring, but whenever we add an ',' as the last item of an initialization list in a test the test fails because of the following error:
Java.lang.AssertionError: Cannot parse
Any ideas how to solve this?

**Comment by yossigil** [comment]
I will close this with @Stav1472

**Comment by dormaayan** [comment]
@yossigil  :
But we are already close to finish.....

**Comment by yossigil** [comment]
Stav: the following lines compile fine on my Eclipse, please check it out carefully, 
```
    int a[] = {1,2,};
    int b[][] = {{1,2,},a,null,};
    int c[][][] = {,};
```

**Comment by Stav1472** [comment]
We know these line are correct, yet we have a problem with the parser, I've sent' you the error message in telegram, can you please help us figure out what is the problem?

**Comment by dormaayan** [comment]
@Stav1472  : I pushed the code @NivShalmon  and me wrote yesterday.
It is in comments now in a wring file named  ` InitializationListRemoveComma.java`
(Don't forget to add it to toolbox.java and that the test are in trimmer and with @ignore...)
Please try to find a way to make the tests work / test it on the plugin itself
then, you can close that issue.

**Comment by yossigil** [comment]
@Stav1472 @dormaayan @M4tt30rru @sakopzon @dngreenst:

Please have a look at the error message that Stav pasted. This is basically my code trying to figure
out what kind of Java snippet you passed it. If it cannot parse it, it dumps all the trials and attempts it made.

**Comment by M4tt30rru** [comment]
I think that the problem takes place when the code to be trimmed is converted to CU (and then to String).
This happens inside the Wrap.find method, and specifically in this instruction:

`  for (final Wrap $ : WRAPS){
        if ($.contains($.intoCompilationUnit(codeFragment).toString(), codeFragment))
        return $;
    }
    azzert.fail("Cannot parse '\n" + codeFragment + "\n********* I tried the following options:" + options(codeFragment));`

In practice, the $.contains method returns FALSE for all the possible wraps available. But the reason why it happens is that $.intoCompilationUnit is altering codefragment. 

As far as I see, it seems that the parser is removing the last comma by itself.

This maybe happens in Funcs.parser, that is called inside MakeAST.makeparser.

If this is the case, I was wondering if there would be possible to set up the Java compiler options taken from the JavaCore class properly - so to avoid that the compiler get rid of the comma by itself. 

I found that there are 466 options, and a number of them are related to the formatting and specifically to commas but, as far as I saw, they imply to add space, not to remove commas.

**Comment by M4tt30rru** [comment]
This link to the JLS might be of some help.

https://docs.oracle.com/javase/specs/jls/se8/html/jls-10.html

Specifically, have a look at chapter 10.6. Array Initializers.

> A trailing comma may appear after the last expression in an array initializer and is ignored.

**Comment by yossigil** [comment]
@Stav1472 FYI

**Comment by Stav1472** [comment]
yes I've noticed, but i can't seem to find a way to test this wring...

**Comment by yossigil** [comment]
You will have to think harder...

**Comment by yossigil** [comment]
You would have to look at how to set the flags.

**Comment by OriRoth** [comment]
@yossigil what more can be done here? How does it related to *preservation of comments*?

**Comment by yuvalsimon** [comment]
I copied some tests from IgnoredTrimmerTest  and Version230 to a new Issue074 test class for this issue, currently ignored.

**Comment by OriRoth** [comment]
Wow nice you manage to track them. I was the one creating `IgnoredTrimmerTest` file using a Python script a while ago... if you have an idea about managing that horrible tests folder that we've got, building a script can be great... a thought for another issue I guess.

**Comment by yossigil** [comment]
Good job @yuvalsimon
